### ISSUE TITLE
Tests failures for perls <= 5.16.3

### ISSUE THREAD

**Issue by dur-randir** [rurban/Cpanel-JSON-XS#43]
```perl
perl -v
This is perl 5, version 16, subversion 3 (v5.16.3) built for x86_64-linux
(with 1 registered patch, see perl -V for more detail)

t/00_load.t ................. ok
t/01_utf8.t ................. 1/13 Argument "2.44_01" isn't numeric in numeric lt (<) at t/01_utf8.t line 33.
t/01_utf8.t ................. ok
t/02_error.t ................ ok
t/03_types.t ................ Failed 8/79 subtests
t/04_dwiw_encode.t .......... ok
t/05_dwiw_decode.t .......... ok
t/06_pc_pretty.t ............ ok
t/07_pc_esc.t ............... ok
t/08_pc_base.t .............. ok
t/09_pc_extra_number.t ...... ok
t/10_pc_keysort.t ........... ok
t/117_numbers.t ............. ok
t/11_pc_expo.t .............. ok
t/12_blessed.t .............. ok
t/13_limit.t ................ ok
t/14_latin1.t ............... ok
t/15_prefix.t ............... ok
t/16_tied.t ................. ok
t/17_relaxed.t .............. ok
t/18_json_checker.t ......... ok
t/19_incr.t ................. ok
t/20_faihu.t ................ ok
t/21_evans.t ................ ok
t/22_comment_at_eof.t ....... ok
t/23_array_ctx.t ............ ok
t/24_freeze_recursion.t ..... ok
t/25_boolean.t .............. 1/13
#   Failed test at t/25_boolean.t line 13.
#          got: 'JSON::PP::Boolean=SCALAR(0x1d81798)'
#     expected: '1'

#   Failed test at t/25_boolean.t line 18.
#          got: 'JSON::PP::Boolean=SCALAR(0x1d817f8)'
#     expected: '0'

#   Failed test 'decode true to !0'
#   at t/25_boolean.t line 30.
#          got: 'JSON::PP::Boolean=SCALAR(0x1d81798)'
#     expected: '1'

#   Failed test 'decode false to !1'
#   at t/25_boolean.t line 31.
# Looks like you failed 4 tests of 13.
t/25_boolean.t .............. Dubious, test returned 4 (wstat 1024, 0x400)
Failed 4/13 subtests
t/52_object.t ............... ok
t/53_readonly.t ............. ok
t/96_interop.t .............. skipped: JSON::XS and JSON required for testing interop
t/96_mojo.t ................. skipped: Mojo::JSON required for testing interop
t/97_unshare_hek.t .......... ok
t/98_56only.t ............... ok
t/99_binary.t ............... ok
t/z_kwalitee.t .............. No subtests run
t/z_leaktrace.t ............. skipped: require Test::LeakTrace
t/z_meta.t .................. skipped: Test::CPAN::Meta 0.12 not available for testing
t/z_perl_minimum_version.t .. skipped: Perl::MinimumVersion 1.20 not available for testing
t/z_pod-coverage.t .......... skipped: Test::Pod::Coverage 1.04 required for testing POD coverage
t/z_pod.t ................... skipped: Test::Pod 1.00 required for testing POD

Test Summary Report
-------------------
t/03_types.t              (Wstat: 0 Tests: 79 Failed: 8)
  Failed tests:  2-4, 6, 8, 10-12
t/25_boolean.t            (Wstat: 1024 Tests: 13 Failed: 4)
  Failed tests:  2, 5, 10-11
  Non-zero exit status: 4
t/z_kwalitee.t            (Wstat: 0 Tests: 0 Failed: 0)
  Parse errors: No plan found in TAP output
Files=40, Tests=31789,  5 wallclock secs ( 1.83 usr  0.09 sys +  4.64 cusr  0.03 csys =  6.59 CPU)
Result: FAIL
Failed 3/40 test programs. 12/31789 subtests failed.
make: *** [test_dynamic] Error 255

**Comment by rurban** [comment]
Yes. I broke master by relying on a working **use overload**. But unfortunately overload was only fixed with 5.18. <5.18 is broken in master.
schmorp has a nice workaround in JSON::XS for this.

But there are many more problems, I'm trying to fix. E.g. bool should be no reference. decode_json("true") needs to error. #41

**Comment by rurban** [comment]
Now fixed in master
